### PR TITLE
added more stack problems and tests

### PR DESCRIPTION
**Pull request by kendricktang**
Towers of hanoi, binary search tree printing,and maximum in O(1) time stack.

### CODE REVIEW

File: array_problems/array_tests/dutchnationalflag_test.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def test_multiple_pivots(self):
 
 def is_sorted(array):
     """Checks to see if an array is sorted"""
+    if len(array) <= 1:
```
Comment (port8080) [inline array_problems/array_tests/dutchnationalflag_test.py]: if not array

File: stackqueue_problems/stackqueue_algs/stack.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        else:
+            self.top = val
+            self.size = 0
+    
+    @classmethod
+    def fromarray(cls, array):
+        """Initializer with array. Highest index will be at the top."""
+        if array:
+            stack = cls(array[0])
+            for i in xrange(1, len(array)):
```
Comment (port8080) [inline stackqueue_problems/stackqueue_algs/stack.py]: write without `xrange`

File: stackqueue_problems/stackqueue_algs/stack.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        """Initializer with single value."""
+        if val:
+            self.top = Node(val)
+            self.size = 1
+        else:
+            self.top = val
+            self.size = 0
+    
+    @classmethod
+    def fromarray(cls, array):
```
Comment (port8080) [inline stackqueue_problems/stackqueue_algs/stack.py]: using `cls` -> :+1:

File: stackqueue_problems/stackqueue_algs/stack.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        """Initializer with array. Highest index will be at the top."""
+        if array:
+            stack = cls(array[0])
+            for i in xrange(1, len(array)):
+                stack.push(array[i])
+            return stack
+        
+    def pop(self):
```
Comment (port8080) [inline stackqueue_problems/stackqueue_algs/stack.py]: What is expected behavior of `pop` on empty stack?

**Comment by port8080** [comment]
Some style comments still, where are you with finding out a python savvy editor?

Your python stye is getting better. Whenever you find yourself typing out xrange, or a while loop, ask yourself if you can write it in a different way.

:shipit:

**Comment by kendricktang** [comment]
Ajit, I just (yesterday) got Vim running with Python compatability so now it has syntax highlighting.

**Comment by kendricktang** [comment]
I have a question about modifying the Stack.fromarray class function. You commented to not use xrange or while loops. Can I still use for loops? e.g. for x in array: stack.push(x)?

File: stackqueue_problems/stackqueue_algs/stack.py (hunk 4 above)
Comment (kendricktang) [inline stackqueue_problems/stackqueue_algs/stack.py]: popping on an empty stack should do nothing. Should it raise an empty stack error?

**Comment by port8080** [comment]
1) awesome
2) yes

**Pull request merged**